src/components/MessagesList/MessagesGroup/Message/MessagePart/Poll.vue

```diff
@@ -52,111 +52,94 @@
-							{{ pollName }}
-						</h2>
-					</div>
-					<div class="poll__summary">
-						{{ pollSummaryText }}
-					</div>
-
-					<!-- options -->
-					<div class="poll__modal-options">
-						<template v-if="checkboxRadioSwitchType === 'radio'">
-							<NcCheckboxRadioSwitch v-for="(option, index) in options"
-								:key="'radio' + index"
-								:checked.sync="vote"
-								class="poll__option"
-								:value="index.toString()"
-								:type="checkboxRadioSwitchType"
-								name="answerType">
-								{{ option }}
-							</NcCheckboxRadioSwitch>
-						</template>
-						<template v-else>
-							<NcCheckboxRadioSwitch v-for="(option, index) in options"
-								:key="'checkbox' + index"
-								:checked.sync="vote"
-								:value="index.toString()"
-								:type="checkboxRadioSwitchType"
-								name="answerType">
-								{{ option }}
-							</NcCheckboxRadioSwitch>
-						</template>
-					</div>
-
-					<div class="poll__modal-actions">
-						<!-- Submit vote button-->
-						<NcButton type="primary" :disabled="!canSubmitVote" @click="submitVote">
-							{{ t('spreed', 'Submit vote') }}
-						</NcButton>
-						<!-- End poll button-->
-						<NcButton v-if="canEndPoll"
-							type="error"
-							@click="endPoll">
-							{{ t('spreed', 'End poll') }}
-						</NcButton>
-					</div>
-				</template>
-
-				<!-- Results page -->
-				<template v-else-if="modalPage === 'results'">
-					<!-- Title -->
-					<div class="poll__header">
-						<PollIcon :size="20" />
-						<h2 class="poll__modal-title">
-							{{ pollName }}
-						</h2>
-					</div>
-					<div class="poll__summary">
-						{{ pollSummaryText }}
-					</div>
-					<div class="results__options">
-						<div v-for="(option, index) in options"
-							:key="index"
-							class="results__option">
-							<div class="results__option-title">
-								<p>
-									{{ option }}
-								</p>
-								<p class="percentage">
-									{{ getVotePercentage(index) + '%' }}
-								</p>
-							</div>
-							<div v-if="getFilteredDetails(index).length > 0 || selfHasVotedOption(index)"
-								class="results__option__details">
-								<PollVotersDetails v-if="details"
-									:details="getFilteredDetails(index)" />
-								<p v-if="selfHasVotedOption(index)" class="results__option-subtitle">
-									{{ t('spreed', 'You voted for this option') }}
-								</p>
-							</div>
-							<NcProgressBar class="results__option-progress"
-								:value="getVotePercentage(index)"
-								size="medium" />
+			<div class="poll__modal" :class="{'poll__modal--closed' : !isPollOpen }">
+				<!-- Title -->
+				<div class="poll__header">
+					<PollIcon :size="20" />
+					<h2 class="poll__modal-title">
+						{{ pollName }}
+					</h2>
+				</div>
+				<div class="poll__summary">
+					{{ pollSummaryText }}
+				</div>
+
+				<!-- options -->
+				<div v-if="modalPage === 'voting'" class="poll__modal-options">
+					<template v-if="checkboxRadioSwitchType === 'radio'">
```

Comment:
```suggestion
					<template v-if="checkboxRadioSwitchType">
```

I'd leave it as a guard, that `checkboxRadioSwitchType` is not undefined